pds_doi_core/outputs/osti_web_parser.py

```diff
@@ -37,6 +39,72 @@ def validate_field_names(self, query_dict):
 
         return o_validated_dict
 
+    def parse_author_names(self,authors_element):
+        """ Given a list of authors element, parse for individual 'first_name', 'last_name' or 'full_name' fields."""
+        o_authors_list = []
+        # If exist, collect all the first_name, last_name or full_name fields into a list of dictionaries.
+        for single_author in authors_element:
+            first_name = single_author.xpath('author/first_name')
+            last_name  = single_author.xpath('author/last_name')
+            full_name  = single_author.xpath('author/full_name')
+            if first_name is not None and len(first_name) > 0 and \
+               last_name  is not None and len(last_name)  > 0:
+                author_dict = {'first_name': first_name[0].text,
+                               'last_name' : last_name[0].text} 
+            if full_name is not None and len(full_name) > 0:
+                author_dict = {'full_name': full_name[0].text}
+            o_authors_list.append(author_dict)
+        return o_authors_list
+
+    def parse_optional_fields(self,io_doi,single_record_element):
+        """ Given a single record element, parse for optional fields that may not be present from the OSTI response:
+                'id', 'site_url', 'doi', 'date_record_added', 'date_record_updated', 'doi_message', 'authors'."""
+
+        o_optional_fields_found = []
+
+        if single_record_element.xpath('id'):
+            io_doi.id = single_record_element.xpath('id')[0].text
+            o_optional_fields_found.append('id')
```

Comment:
we could have a debug statement instead saying "adding optional field id to DOI". That would be more explicit than having a variable just to handle the debug message at the end.

Reply:
Per your recommendation, the debug statements have been added in favor of having a list and then adding to it.

Reply:
Thank you Qui